packages/edge-login-ui-rn/src/components/themed/PasswordStatus.js

```diff
@@ -0,0 +1,121 @@
+// @flow
+
+import * as React from 'react'
+import { View } from 'react-native'
+import { cacheStyles } from 'react-native-patina'
+import FontAwesome from 'react-native-vector-icons/FontAwesome'
+import SimpleLineIcons from 'react-native-vector-icons/SimpleLineIcons'
+
+import s from '../../common/locales/strings.js'
+import { type PasswordStatusState } from '../../reducers/PasswordStatusReducer'
+import { type RootState } from '../../types/ReduxTypes'
+import { fixSides, mapSides, sidesToMargin } from '../../util/sides'
+import { connect } from '../services/ReduxStore'
+import {
+  type Theme,
+  type ThemeProps,
+  withTheme
+} from '../services/ThemeContext'
+import { EdgeText } from './EdgeText'
+import { Fade } from './Fade'
+
+type OwnProps = {
+  marginRem?: number[] | number
+}
+
+type StateProps = {
+  status: PasswordStatusState | null
+}
+type Props = OwnProps & StateProps & ThemeProps
+
+const PasswordStatusComponent = ({ status, marginRem, theme }: Props) => {
+  const styles = getStyles(theme)
+  const spacings = sidesToMargin(mapSides(fixSides(marginRem, 0.5), theme.rem))
+
+  if (status === null) return null
+
+  const { list, passed } = status
+
+  return (
+    <View style={[styles.container, spacings]}>
+      <View style={styles.top}>
+        <View style={styles.iconsContainer}>
+          <View style={styles.icon}>
+            <Fade visible={passed}>
+              <FontAwesome
+                name="check-circle"
+                size={theme.rem(1.1)}
+                color={theme.positiveText}
+              />
+            </Fade>
+          </View>
+          <View style={styles.icon}>
+            <Fade visible={!passed}>
+              <SimpleLineIcons
+                name="info"
+                size={theme.rem(1.1)}
+                color={theme.warningText}
+              />
+            </Fade>
+          </View>
+        </View>
+        <EdgeText style={[styles.message, passed && styles.passed]}>
+          {s.strings.password_requirements}
+        </EdgeText>
+      </View>
+      {list.map(({ title, value }) => (
+        <EdgeText
+          key={title}
+          style={[styles.case, value && styles.passed]}
+        >{`\u2022 ${title}`}</EdgeText>
+      ))}
+    </View>
+  )
+}
+
+const getStyles = cacheStyles((theme: Theme) => ({
+  container: {
+    borderWidth: 0.25,
+    borderRadius: theme.rem(0.25),
+    borderColor: theme.warningText,
```

Comment:
The `borderColor` should switch to `theme.positiveText` in case `passed` is true.